Py 105 prepare code for multiple api versions

gabrys opened the pull request:
## Summary by Sourcery

Refactor the neptune-fetcher codebase to centralize and decouple public and internal logic in preparation for multiple API versions support. Move filter and exception classes out of the alpha module into shared internal modules, introduce resolution utilities in alpha to translate public arguments to internal types, and update all references and tests accordingly.

Enhancements:
- Introduce a new internal filters package and migrate all filter classes and logic from the alpha layer into it
- Move exception definitions out of the alpha module into a centralized top-level exceptions module
- Add an alpha/_internal module with resolve_* functions to convert public filter and attribute inputs into internal representations
- Refactor all alpha public API functions (list, fetch, download) to delegate to internal composition and retrieval modules via resolved internal filters and attributes
- Update type hints across the codebase to use built-in generics (e.g. list[str])

Tests:
- Update unit and end-to-end tests to import and use the new internal filter and attribute classes
- Adjust conftest fixtures and test imports to reflect relocated modules and updated APIs

Chores:
- Clean up unused alpha internal modules and adjust imports everywhere to point to the new internal packages
- Add __all__ exports in moved modules for proper public interface control

pitercl commented on tests/e2e/conftest.py:
```diff
@@ -1,45 +1,59 @@
+    get_config_and_token_urls,
+)
+
+# TODO: un-alphafy
```
Remove?

pitercl commented on src/neptune_fetcher/internal/filters.py:
```diff
@@ -0,0 +1,393 @@
+
+
+class _Filter(ABC):
+    """Filter used to specify criteria when fetching experiments or runs.
```
We should probably at least remove mentions of `alpha` in the docs, if we want to keep them (which maybe makes sense? Not sure, tbh).

gabrys replied:
Removed alpha, kept the rest of the docs.

pitercl commented on src/neptune_fetcher/exceptions.py:
```diff
@@ -0,0 +1,248 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# src/neptune_fetcher/alpha/internal/exception.py
```
Remove the line

pitercl commented on tests/e2e/conftest.py:
```diff
@@ -1,45 +1,57 @@
-    ReadOnlyProject,
-    ReadOnlyRun,
+from neptune_fetcher import ReadOnlyProject
+from neptune_fetcher.alpha.filters import Filter
```
Switch to internal

pitercl commented on tests/e2e/internal/retrieval/test_attributes.py:
```diff
@@ -9,26 +9,26 @@
 import pytest
 
-from neptune_fetcher.alpha.exceptions import NeptuneProjectInaccessible
 from neptune_fetcher.alpha.filters import AttributeFilter
```
Switch to internal

pitercl commented on tests/e2e/internal/retrieval/test_search.py:
```diff
@@ -9,18 +9,18 @@
 import pytz
 
-from neptune_fetcher.alpha.exceptions import NeptuneProjectInaccessible
 from neptune_fetcher.alpha.filters import (
```
Switch to internal

gabrys commented:
I addressed all the comments from @pitercl . Also, to verify the extraction of `internal` from `alpha`, I ran e2e tests on the copy of the code with all `alpha` code removed: (42fb546): https://github.com/neptune-ai/neptune-fetcher/actions/runs/15218162748 the tests passed which suggests no non-alpha code or test depends on `alpha`.
